tb_plugin/torch_tb_profiler/plugin.py

```diff
@@ -222,15 +222,19 @@ def trace_route(self, request):
 
         run = self._get_run(name)
         profile = run.get_profile(worker)
-        raw_data, includes_gpu_metrics = self._cache.read(profile.trace_file_path)
+        file_with_gpu_metrics = profile.gpu_metrics_file_mapping.get(profile.trace_file_path)
+        if file_with_gpu_metrics is None:
+            raw_data = self._cache.read(profile.trace_file_path)
+        else:
+            with open(file_with_gpu_metrics, 'rb') as f:
+                raw_data = f.read()
 
         if not profile.has_kernel:  # Pure CPU.
             if not profile.trace_file_path.endswith('.gz'):
                 raw_data = gzip.compress(raw_data, 1)
```

Comment:
move the logic into the if file_with_gpu_metrics. something like:

if file_with_gpu_metrics is None:
    raw_data = self._cache.read(profile.trace_file_path)
    if profile.has_kernel:
        raw_data = self._append_gpu_metrics(profile, raw_data)
    else:
        if not profile.trace_file_path.endswith('.gz'):
            raw_data = gzip.compress(raw_data, 1)
else:
    with open(file_with_gpu_metrics, 'rb') as f:
        raw_data = f.read()

Reply:
Cool! Thanks. Done.